ODIN_II/SRC/OdinGridAnalyzer.cpp

```diff
@@ -0,0 +1,325 @@
+    int priorityOfTheDesiredBlock = findDesiredBlockPriority(layout, typeTag);
+    std::cout << "Desired block's priority is: " << priorityOfTheDesiredBlock << std::endl;
+    std::cout << "BlockDefsAre:" << std::endl;
+    for (pointer = layout.loc_defs.begin(); pointer < layout.loc_defs.end(); pointer++) {
+        // Case of == is the tile type we are looking for, and a clash
+        // if there are collisions, vpr would fail anyway
+        if (priorityOfTheDesiredBlock <= pointer->priority) {
+            std::cout << pointer->block_type << ":" << std::endl;
+            ssize_t indexOfTile = findIndexOfAssociatedTileType(tileTypes, pointer->block_type);
+            if (indexOfTile == -1) {
+                std::cout << " Was not able to find a tile" << std::endl;
+                flush(std::cout);
+                exit(6);
+            }
+            t_physical_tile_type& tileTypeOfABlockWithHigherPriority = tileTypes[indexOfTile];
+            int w = 0;
+            int h = 0;
+            std::cout << "Width of a block is:\t" << tileTypeOfABlockWithHigherPriority.width << "\tHeight of a block is:"
+                      << tileTypeOfABlockWithHigherPriority.height << std::endl;
+            w = tileTypeOfABlockWithHigherPriority.width;
+            h = tileTypeOfABlockWithHigherPriority.height;
+
+            fillGridWithBlockType(grid, &(*pointer), tileTypes, indexOfTile, grid_width, grid_height);
+        }
+    }
+    std::cout << "Grid is:" << std::endl;
+    for (size_t x = 0; x < grid_width; x += 1) {
+        for (size_t y = 0; y < grid_height; y += 1) {
+            if (grid[x][y].first != nullptr) {
+                std::cout << "(" << grid[x][y].first << "," << grid[x][y].second << ")\t";
+            } else {
+                std::cout << "( u, -1), ";
+            }
+        }
+        std::cout << std::endl;
+    }
+    int count = 0;
+    std::cout << "Grid is:" << std::endl;
+    for (size_t x = 0; x < grid_width; x += 1) {
+        for (size_t y = 0; y < grid_height; y += 1) {
+            if (grid[x][y].first != nullptr) {
+                std::string stringToCompare = std::string(grid[x][y].first);
+                if (stringToCompare.compare(typeTag) == 0)
+                    count = count + 1;
+            }
+        }
+    }
+    //TODO consider different mults
+    return count;
+}
+void OdinGridAnalyzer::fillGridWithBlockType(std::vector<std::vector<std::pair<char*, int>>>& grid,
+                                             t_grid_loc_def* grid_loc_def,
+                                             std::vector<t_physical_tile_type> tileTypes,
+                                             ssize_t indexOfTile,
+                                             int grid_width,
+                                             int grid_height) {
+    t_physical_tile_type* type = &(tileTypes[indexOfTile]);
+    /**
+     * This code is a lightweight version of vpr/src/base/SetupGrid.cpp
```

Comment:
I wonder if you could just hijack the function thats there itself rather than a rewirte? how much heavier is it?

Reply:
There are a lot of includes that are required and the other one is more heavyweight

Reply:
>  one is more heavyweight

but how much... is it gonna introduce a 2 minute slowdown? even 5 seconds slowdown is fine by me. maybe just declare it as extern first and we can compile against it. I wonder if its compiled as a lib in vtr. I would rather not have to maintain something equivalent for the sake of it or even some arginal gains.

Reply:
it is part of libvpr... as long as there is no conflict, it should be easy to include in CMake to build Odin. Going foward, Odin will probably start consuming more pieces of vpr, so I don't think its a terrible idea.